### DISCUSSION TITLE
http services have no way to render using webview

### DISCUSSION THREAD

**Discussion by hanchenglong001** [flet-dev/flet#3133] [Q&A] [answered]
### **Description**
Packaging HTTP services using IP into Android apps, unable to display using webview

### **Code example to reproduce the issue:**
```
import time

import flet as ft
import threading
from flask import Flask,send_file

PORT=5448

app = Flask(__name__)
@app.route('/')
def hello_world():
    return send_file("assets/test.html")


def run_flask_in_thread():
    app.run(host='0.0.0.0', port=PORT)



def main(page: ft.Page):
    page.title = "测试工具"
    flask_thread = threading.Thread(target=run_flask_in_thread)
    flask_thread.start()
    page.navigation_bar = ft.NavigationBar(
        destinations=[
            ft.NavigationDestination(icon=ft.icons.EXPLORE, label="首页")
        ]
    )
    wv = ft.WebView(
        url=f"http://127.0.0.1:{PORT}/",
        expand=True,
        on_page_started=lambda _: print("Page start"),
        on_page_ended=lambda _: print("Page ended"),
        on_web_resource_error=lambda e: print("Page error:", e.data),
    )
    page.add(wv)


ft.app(target=main)
```

**images example to reproduce the issue:**

![image](https://github.com/flet-dev/flet/assets/56539098/79244081-6351-4173-a77b-3215adac6de3)

**Describe the results you received:**
err_cleartext_not_permitted

**Describe the results you expected:**
Can access the service normally. And render the HTML page。

**Operating system**:
Windows
python3.9
Android12

requirements.txt
```
flet==0.22.*
requests==2.31.0
flask==3.0.3
```

**Comment by FeodorFitsner** [comment]
This could be helpful: https://kinsta.com/knowledgebase/net-err_cleartext_not_permitted/

**Reply by hanchenglong001** [reply]
Thank you for your answer. But I packed it through the order you provided.
flet build apk  -v --no-android-splash --include-packages flet_webview
How do I modify the AndroidManifest.xml file?

**Reply by hanchenglong001** [reply]
Our problem's solved. 
I decompiled and then modified the package.

**Reply by FeodorFitsner** [reply]
Could you please share what was added/changed to `AndroidManifest.xml` to fix the problem?

**Reply by hanchenglong001** [reply]
![image](https://github.com/flet-dev/flet/assets/56539098/28109062-3f74-4625-9bb2-0a177e0b2b42)

**Reply by RRozi** [reply]
I'm making a WebApp for Telegram - the same error, what should I do?

**Reply by PhunkyBob** [reply]
> Our problem's solved. I decompiled and then modified the package.

Instead of decompile / compile / sign, you can:
- Fork the template at https://github.com/flet-dev/flet-build-template (don't fork the `main` branch, but a more up-to-date one).
- Go to `flet-build-template\{{cookiecutter.out_dir}}\android\app\src\main` and modify the `AndroidManifest.xml` file.
- When you build your APK, use the `--template` option and give the path to your modified template folder.

Your APK will be as you wish and already signed.

**Reply by ndonkoHenri** [reply, marked as answer]
> what should I do?

Add this to the `pyproject.toml`:
```toml
[tool.flet.android.manifest_application]
usesCleartextTraffic = "true"
```
